src/CMakeLists.txt

```diff
@@ -1341,6 +1341,24 @@ if(HAVE_CLOCK_GETTIME)
    list( APPEND DEFINES HAVE_CLOCK_GETTIME )
 endif()
 
+set( AUDACITY_LIBRARIES
+# A sub-sequence of what is in libraries/CMakeLists.txt :
+   lib-string-utils-interface
+   lib-uuid-interface
+   lib-screen-geometry-interface
+   lib-project-rate-interface
+   lib-ffmpeg-support-interface
+   lib-theme-resources-interface
+   lib-track-interface
+   lib-transactions-interface
+   lib-network-manager-interface
```

Comment:
I think it is worth adding a comment, that this list is purely for building the graph.

Also, this looks very fragile. It is not documented, I do expect it to break with the next library added.

Reply:
If "break" means the build fails, then you add a line there to fix the build, before you push, and maybe remove older lines that are unnecessary because they are now transitive dependencies.  That is what I intend.

What is wrong with that?

Reply:
The build won't fail and that is the problem. Or do I miss something?

Reply:
You think the build won't fail when it should fail?

But (trust me) the build does fail in my monster branch when I extract a library but neglect to update this list, because Audacity tries to `#include` a header file from the new library, which is not found.

That build breakage is "good," early fragility that forces a fix at compile time (even before link time), not "bad," late fragility that lets the program build and run but then fail at startup time.

Did you also read the PR comment too, explaining the advantages I seek from this.

Reply:
Well then I really missed something. I will try to figure out tomorrow

Reply:
The libraries in the list are given to `target_link_libraries` which affects more than the link step:  it affects also the `-I` options at compile time.

Reply:
It will make sense in the morning, I trust.  Спокойной ночи.

Reply:
I will recheck, it is not clear to me how the build works with networking disabled in this case.

Reply:
Doesn't build, so I was right!

```
LINK : fatal error LNK1104: cannot open file 'lib-network-manager-interface.lib' [C:\devel\projects\audacity\Paul\tie\b
uild\src\Audacity.vcxproj]```

Reply:
That will be easily fixed by adding one to the list of direct dependencies.  But didn’t my CI run complete?  Are you using the same CMake options as in the CI build?

Even if I can and do fix the PR with one line, why is it a reason not to approve this PR, that the build might break because one line is missing?

Reply:
https://github.com/audacity/audacity/actions/runs/1574761410

That build succeeded yesterday.

Reply:
The CI builds do turn on networking, but I also needed to make the build work if you do not enable it, as must have been so for your local builds but not mine.  There is another commit in the branch.  I will squash and merge if you approve.

To see why I want the change, look at this generated graph with the change:
![modules dot](https://user-images.githubusercontent.com/11670369/145984644-f00f1a32-c409-4932-b180-90dc35374321.gif)

Reply:
Compare without the change.  See a lot of redundant edges that don't change the transitive closure and make it harder to understand.

![modules dot](https://user-images.githubusercontent.com/11670369/145984822-f75b6d4a-a826-469a-b56c-4d4487b6f71b.gif)

Reply:
Compare without:
![modules dot](https://user-images.githubusercontent.com/11670369/145985159-35422e57-3666-4270-a50d-a067fa643359.gif)

Reply:
I build with default CMake options which are different from CI options. It is required that Audacity can be built without networking (and it was your call actually to disable networking by default).

Reply:
There are fewer edges in the first picture, but the transitive closure is the same.  I think the first can be easier to understand.  The difference will be more pronounced as more libraries are extracted.

Reply:
> I build with default CMake options which are different from CI options. It is required that Audacity can be built without networking (and it was your call actually to disable networking by default).

Indeed, but I often make local builds with everything enabled to make sure I don't break things that CI will build.

This was easily fixed with the second commit in this branch.